src/zarr/errors.py

```diff
@@ -52,14 +57,19 @@ def __init__(self, *args: Any) -> None:
             super(BaseZarrError, self).__init__(_msg.format(*args))
 
 
-@deprecated("Use NodeNotFoundError instead.", category=None)
-class PathNotFoundError(BaseZarrError):
-    # Backwards compatibility with v2. Superseded by NodeNotFoundError.
-    ...
-
+class GroupNotFoundError(NodeNotFoundError):
+    """
+    Raised when a group isn't found at a certain path.
+    """
 
-class NodeNotFoundError(PathNotFoundError):
-    """Raised when an array or group does not exist at a certain path."""
+    def __init__(self, *args: Any) -> None:
+        if len(args) == 1:
+            # Pre-formatted message
+            super(BaseZarrError, self).__init__(args[0])
+        else:
+            # Store and path arguments - format them
+            _msg = "No group found in store {!r} at path {!r}"
+            super(BaseZarrError, self).__init__(_msg.format(*args))
```

Comment:
in a separate PR we should remove the pre-formatting of exception messages, and make every exception just take a single string